pkg/domain/infra/abi/play.go

```diff
@@ -125,6 +125,59 @@ func (ic *ContainerEngine) createServiceContainer(ctx context.Context, name stri
 	return ctr, nil
 }
 
+func (ic *ContainerEngine) prepareAutomountImages(ctx context.Context, forContainer string, annotations map[string]string) ([]*specgen.ImageVolume, error) {
+	volMap := make(map[string]*specgen.ImageVolume)
+
+	ctrAnnotation := define.KubeImageAutomountAnnotation + "/" + forContainer
+
+	automount, ok := annotations[ctrAnnotation]
+	if !ok || automount == "" {
+		return nil, nil
+	}
+
+	for _, imageName := range strings.Split(automount, ";") {
+		img, fullName, err := ic.Libpod.LibimageRuntime().LookupImage(imageName, nil)
+		if err != nil {
+			return nil, fmt.Errorf("image %s from container %s does not exist on local storage, cannot automount: %w", imageName, forContainer, err)
+		}
+
+		logrus.Infof("Resolved image name %s to %s for automount into container %s", imageName, fullName, forContainer)
+
+		inspect, err := img.Inspect(ctx, nil)
+		if err != nil {
+			return nil, fmt.Errorf("cannot inspect image %s to automount into container %s: %w", fullName, forContainer, err)
+		}
+
+		volumes := inspect.Config.Volumes
+		if len(volumes) != 1 {
```

Comment:
The data volume container images can have a single VOLUME directive as a list of volumes. For example,
`VOLUME ["/path1/to/vol1", "/path1/to/vol2", "/path2/to/vol3" ]`

Reply:
In that case, do you want to mount the image to every path given as a VOLUME?

Reply:
Right, all the defined volumes to the respective path.

Reply:
Also, just to be clear it’s not the entire image that we want to mount. Just the specific volume paths in the image.

Reply:
So, to be clear, if the volume is `/path/to/vol1`, you want to mount `/path/to/vol1` in the image to `/path/to/vol1` in the container?

Reply:
Alright. Based on this, new requirements are:
* The image must already exist in storage.
* The image must have at least 1 Volume directive.
* The volume path in the image will be mounted to the same path in the container; IE, if the image specifies a volume at `/my/volume/a`, the content in the image at `/my/volume/a` will be made available in the container at `/my/volume/a`
* More than one image can be specified, but the paths they mount to (the Volume directives) must be different for all volumes in all images mounted into a single container.
* The images are always mounted read-only.
* Images to mount are defined in the annotation "io.podman.annotations.kube.image.automount/$ctrname" as a semicolon-separated list. They are mounted into a single container in the pod, not the whole pod.

Reply:
Correct. Thanks.

Reply:
On the requirement "but the paths they mount to (the Volume directives) must be different for all volumes in all images mounted into a single container", I was thinking that instead of aborting the operation can we just keep overriding the mount point with the next one in the order in case of conflict? Podman can log a warning message though.

Reply:
That should be workable.

Reply:
Thanks Matt.